File: AC-ABC/abc122b.go
package main

import "fmt"

func main() {
	var s string
	var max, tmp int = 0, 0
	fmt.Scan(&s)
	for _, c := range s {
		switch string(c) {
		case "A", "C", "G", "T":
			tmp++
		default:
			if max < tmp {
				max = tmp
				tmp = 0
			}
		}
	}
	if max < tmp {
		max = tmp
		tmp = 0
	}
	fmt.Println(max)
}

File: AC-ABC/abc122b.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  string s;
  cin >> s;
  int max, tmp = 0;
  for (int i = 0; i < s.length(); i++) {
    switch (s[i]) {
      case 'A':
      case 'C':
      case 'G':
      case 'T':
        tmp++;
        break;
      default:
        if (max < tmp) {
          max = tmp;
          tmp = 0;
        }
    }
    cout << "tmp:" << tmp << endl;
  }
  if (max < tmp) {
    max = tmp;
    tmp = 0;
  }
  cout << max << endl;
}
